src/python/pants/backend/python/lint/pydocstyle/rules.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+from typing import Tuple
+
+from pants.backend.python.lint.pydocstyle.subsystem import Pydocstyle, PydocstyleFieldSet
+from pants.backend.python.subsystems.setup import PythonSetup
+from pants.backend.python.util_rules import pex
+from pants.backend.python.util_rules.interpreter_constraints import InterpreterConstraints
+from pants.backend.python.util_rules.pex import PexRequest, VenvPex, VenvPexProcess
+from pants.core.goals.lint import LintResult, LintTargetsRequest, Partitions
+from pants.core.util_rules.config_files import ConfigFiles, ConfigFilesRequest
+from pants.core.util_rules.partitions import Partition
+from pants.core.util_rules.source_files import SourceFiles, SourceFilesRequest
+from pants.engine.fs import Digest, MergeDigests
+from pants.engine.process import FallibleProcessResult
+from pants.engine.rules import Get, MultiGet, collect_rules, rule
+from pants.util.logging import LogLevel
+from pants.util.strutil import pluralize
+
+
+class PydocstyleRequest(LintTargetsRequest):
+    field_set_type = PydocstyleFieldSet
+    tool_subsystem = Pydocstyle
+
+
+def generate_argv(source_files: SourceFiles, pydocstyle: Pydocstyle) -> Tuple[str, ...]:
+    args = []
+    if pydocstyle.config is not None:
+        args.append(f"--config={pydocstyle.config}")
+    args.extend(pydocstyle.args)
+    args.extend(source_files.files)
+    return tuple(args)
+
+
+@rule
+async def partition_pydocstyle(
+    request: PydocstyleRequest.PartitionRequest[PydocstyleFieldSet],
+    pydocstyle: Pydocstyle,
+    python_setup: PythonSetup,
+) -> Partitions[PydocstyleFieldSet, InterpreterConstraints]:
+    if pydocstyle.skip:
+        return Partitions()
+
+    constraints_to_field_sets = InterpreterConstraints.group_field_sets_by_constraints(
+        request.field_sets, python_setup
+    )
```

Comment:
Do you know how Pydocstyle handles interpreter constraints? Some tools don't care what Python interpreter version is used, like isort. Some do, like Flake8.

Reply:
There's no mention of Python version specific behavior in the documentation. Most likely, there won't be any difference. However, I still found this to be very useful to be able to differentiate between the Python versions. For example, one may want to apply different rules to legacy Python 2 code and another one for a new Python 3 code. I'd love to keep the constraints. WDYT?

Reply:
I think in that case we'd partition by config, not by ICs. As it stands because different ICs could be anywhere, we're maybe-partitioning-by-config-maybe-not.

So I think this shouldn't do any partitioning (for consistency with other tools) and we can come back and partition by config when we do that for everyone.

Reply:
> So I think this shouldn't do any partitioning

Yeah, I agree with starting without partitioning. Keep it simple. We can add partitioning later, but it's harder to take it away.

Reply:
Thank you! Partitions have been removed.